### PR TITLE
Fix error on missing `ingress.paths` value

### PR DESCRIPTION
**Pull request by 11000100111000**
# What this PR does / why we need it
This PR fixes the need for specifying `master.ingress.paths` even if no paths need to be overwritten.
This is done by adding an additional check to see if  `master.ingress.paths` is even defined before taking the length

# Which issue this PR fixes

# Special notes for your reviewer

# Checklist

- [x] [DCO](https://github.com/jenkinsci/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] CHANGELOG.md was updated

### CODE REVIEW

**Comment by 11000100111000** [comment]
@torstenwalter There is still the inconsistency between `ingress.paths` and `secondaryingress.paths`. However i don't this there is a good way to change that since it currently makes secondaryingress easier to configure for it's intended purpose. Do you have any opinion on that?

Also i again can't figure out where i broke the tests, locally doesn't work with or without my changes if i copy everything in `set` to a file and template using that file. Somehow `master.ingress` doesn't work for me.

**Comment by 11000100111000** [comment]
@torstenwalter could you please take a look at this PR since today is the last day of hacktoberfest :sparkles:

File: charts/jenkins/tests/jenkins-master-ingress-test.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-suite: Controller Secondary Ingress
+suite: Controller Primary Ingress
```
Comment (torstenwalter) [inline charts/jenkins/tests/jenkins-master-ingress-test.yaml:1]: Good point.

File: charts/jenkins/values.yaml [hunk 2]
```diff
@@ -391,7 +391,7 @@ master:
   ingress:
     enabled: false
     # Override for the default paths that map requests to the backend
-    paths: []
+    # paths:
```
Comment (torstenwalter) [inline charts/jenkins/values.yaml:394]: I'd prefer to leave the empty list as default. That allows auto completion.

**Review by torstenwalter — COMMENTED** [review]
How can I reproduce the error you are trying to fix?

Just a side not: It makes it easier if you create a branch on your repository when creating PRs.

**Comment by 11000100111000** [comment]
The error was that if you don't specify it in the values you get a error because you use len on it, if you want it so stay as `[]` it shouldn't technically be an issue.
It's still weird that primary ingress needs it while it is optional for the secondary ingress

**Comment by torstenwalter** [comment]
I think that's somehow by design. For the primary ingress we assume that wou want to expose Jenkins.
The secondary ingress is used for webhooks depending on which version control system you are using this might be different.

- GitHub
  https://plugins.jenkins.io/github/
  `$JENKINS_BASE_URL/github-webhook/`

- BitBucket
  https://plugins.jenkins.io/bitbucket/
  `JENKINS_URL/bitbucket-hook/`

- GitLab
  https://github.com/jenkinsci/gitlab-plugin#configuring-global-authentication
  `https://USERID:APITOKEN@JENKINS_URL/project/YOUR_JOB`

**Comment by 11000100111000** [comment]
Yeah but i think it's weird to force a empty array there if you want the "default" configuration, Also it's kinda better for trying things out since this:
```yaml 
 paths:
    # - backend:
    #     serviceName: ssl-redirect
    #     servicePort: use-annotation
```

would create an error on main vs the default configuration on my branch

File: charts/jenkins/values.yaml (hunk 2 above)
Comment (11000100111000) [reply charts/jenkins/values.yaml]: Fixed :)

**Comment by torstenwalter** [comment]
For me the default argument is not that strong as it is defaults are provided by values.yaml in the chart.

I am all in on the point that it makes it easier to experiment with the settings as you could copy relevant section from values.yaml of the chart to your own and simply remove comments.

So let's add a unit test which passes on your branch, but not in main. That helps us ensure that we keep the behavior.

**Comment by 11000100111000** [comment]
Alright! Will add a Test later today :+1:

**Comment by 11000100111000** [comment]
why did it close this pr? ^^

**Comment by 11000100111000** [comment]
I've added a test and created two PRs on my fork to see the result of the test [Changes](https://github.com/11000100111000/helm-charts/pull/4) vs [Main](https://github.com/11000100111000/helm-charts/pull/3)

**Review by torstenwalter — APPROVED** [review]
Thank you. That makes it much cleaner. I suspect that we have other places with empty arrays which could benefit from similar improvements.

**Comment by admoriarty** [comment]
@11000100111000 @torstenwalter  I'm not sure if this introduced a bug or I'm not understanding the changes, but where I used to get by with just;

```
master:
  ingress:
    path: my-jenkins
```

I know need to do the following hardcoding to get a complete ingress;

```
master:
  ingress:
     paths:
          - backend:
              serviceName: my-jenkins
              servicePort: 8080
      path: "/my-jenkins"
```

It seems that the floating "path" insertion below is probably trapped in an else block it shouldn't be in, it seems like it should be moved up into the if block for empty "paths" handling? Which was the initial intention of https://github.com/jenkinsci/helm-charts/commit/c3136130edda539467d0024bb3bbbb254697f1ba by the looks of it?

```
{{- if .Values.master.ingress.path }}
        path: {{ .Values.master.ingress.path }}
{{- end -}}
```

**Comment by 11000100111000** [comment]
I think i get what you mean ^^
It seems like i introduced the bug(i didn't notice that the if was nested), do you already have a fix and want to submit it or should i fix it(albeit later today or tomorroww)?

**Comment by admoriarty** [comment]
I'll have a go later on, wanted to check first, thanks

**Comment by admoriarty** [comment]
@11000100111000 Sorry but I never got back to this, do you know if it has been patched (lots of movement from 2.x.to 3.x) ?

**Comment by 11000100111000** [comment]
> @11000100111000 Sorry but I never got back to this, do you know if it has been patched (lots of movement from 2.x.to 3.x) ?

I don't know

**Pull request merged**